Gentoo specific patch, that switches off GoogleTest and Google Benchmark
download to build and run the tests but enables the build within native
Gentoo packages.

diff '--color=auto' -urw a/tests/CMakeLists.txt b/tests/CMakeLists.txt
--- a/tests/CMakeLists.txt	2025-11-18 00:58:12.572764578 +0300
+++ b/tests/CMakeLists.txt	2025-11-18 01:12:38.676143728 +0300
@@ -2,42 +2,9 @@
 
 project(${CMAKE_PROJECT_NAME}Tests CXX)
 
-# Set up GoogleTest and Benchmark
-include(FetchContent)
-
-FetchContent_Declare(
-  benchmark
-  GIT_REPOSITORY https://github.com/google/benchmark.git
-  GIT_TAG v1.9.4
-  GIT_SHALLOW TRUE
-)
-
-FetchContent_Declare(
-  googletest
-  GIT_REPOSITORY https://github.com/google/googletest.git
-  GIT_TAG v1.17.0
-  GIT_SHALLOW TRUE
-)
-
-set(BENCHMARK_DOWNLOAD_DEPENDENCIES OFF CACHE BOOL "" FORCE)
-set(BENCHMARK_ENABLE_INSTALL OFF CACHE BOOL "" FORCE)
-set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "" FORCE)
-set(BENCHMARK_INSTALL_DOCS OFF CACHE BOOL "" FORCE)
-
-FetchContent_MakeAvailable(benchmark)
-
-# For Windows: Prevent overriding the parent project's compiler/linker settings
-set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-
-# Disable building GMock
-set(BUILD_GMOCK OFF CACHE BOOL "" FORCE)
-
-# Do not install GTest 
-set(INSTALL_GTEST OFF CACHE BOOL "" FORCE)
-
-FetchContent_MakeAvailable(googletest)
-
-set_target_properties(gtest gtest_main benchmark benchmark_main PROPERTIES FOLDER "tests/third-party")
+# Detect GoogleTest and Benchmark
+find_package(benchmark REQUIRED)
+find_package(GTest REQUIRED)
 
 foreach(executable dbf_test sbn_test shp_test shp_bench)
   add_executable(${executable} ${PROJECT_SOURCE_DIR}/${executable}.cc)
